CACHING.md

```diff
@@ -0,0 +1,351 @@
+
+```yaml
+steps:
+  - uses: actions/checkout@v3
+  .
+  . // restore if need be
+  .
+  - name: Build
+    run: /build.sh
+  - uses: actions/cache/save@v3
+    if: always() // or any other condition to invoke the save action
+    with:
+      path: path/to/dependencies
+      key: ${{ runner.os }}-${{ hashFiles('**/lockfiles') }}
+```
+
+## Restoring and saving caches
+
+### Restoring and saving cache using a single action
+
+The [cache](https://github.com/actions/cache) action allows caching dependencies and restoring them using a single action. It has a `main` step and a `post` step. In the `main` step, the cache is restored if it exists for the input `key`, `path` combination. If cache is not found for the given `key` input, then cache is restored using `restore keys`. If the cache doesn't exist or is restored using `restore-keys`, the cache is saved in the `post` step of this action.
+
+```yaml
+  - uses: actions/cache@v3
+    with:
+      path: |
+        path/to/dependencies
+        some/other/dependencies
+      key: ${{ runner.os }}-${{ hashFiles('**/lockfiles') }}
+```
+
+The `cache` action provides one output `cache-hit` which is set to `true` when cache is restored using primary key and `false` when cache is restored using `restore-keys` or no cache is restored.
+
+### Download remaining dependencies in case of cache miss
+
+In case cache gets download using restore keys, there's a chance that some dependencies might be missing. It might also be possible that no cache was restored because there was no match. In such cases, the output `cache-hit` is set to `false`. We can make use of this output to download the remaining dependencies.
+
+```yaml
+  - uses: actions/cache@v3
+    with:
+      path: |
+        path/to/dependencies
+        some/other/dependencies
+      key: ${{ runner.os }}-${{ hashFiles('**/lockfiles') }}
+  - name: Install Dependencies
+    if: steps.cache.outputs.cache-hit != 'true'
+    run: ./install-dependencies.sh
+```
+
+### Using combination of restore and save actions
+
+```yaml
+  - uses: actions/cache/restore@v3
+    with:
+      path: |
+        path/to/dependencies
+        some/other/dependencies
+      key: ${{ runner.os }}-${{ hashFiles('**/lockfiles') }}
+  - name: build
+    run: ./install.sh
+  - uses: actions/cache/save@v3
+    with:
+      path: |
+        path/to/dependencies
+        some/other/dependencies
+      key: ${{ runner.os }}-${{ hashFiles('**/lockfiles') }}
+```
```

Comment:
This looks like should already be covered in `README as it talking about basics of how to use cache and other new actions.

Reply:
Its not there in README, I'll shift it there.

Reply:
Moved.